### ISSUE TITLE
✨ New `orNull(Byte)` and `orThrow(Byte)` in `Zero.Companion`

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#685]
## 📝 Description

We want to introduce the `orNull(Byte)` and the `orThrow(Byte)` factory functions to the `Zero.Companion` type for creating an instance of `Zero` in an expressive way. Also, `null` values not being well supported by the Java compiler, the `orNull(Byte)` function shouldn't be available for Java, but only for Kotlin platforms. The `orThrow(Byte)` function should be available for all Kotlin platforms and Java.

```kotlin
fun Zero.Companion.orNull(number: Byte): Zero?
fun Zero.Companion.orThrow(number: Byte): Zero

// Before
Zero.fromByte(0)
Zero.fromByteOrNull(0)
// After
Zero.orThrow(0)
Zero.orNull(0)
```

After adding these functions, we want deprecate the `fromByte` and `fromByteOrNull` functions from the `Zero.Companion` **experimental** type with an error level for using these new factory functions.

## 🔗 Dependencies

This issue is blocked by the following items:

- [x] #681 
- [x] #683 

## ✅ Checklist

- [ ] ✨ Add the `Zero.Companion.orThrow(Byte)` function for Kotlin and Java with tests, documentation and samples, then update the unreleased changelog.
- [ ] ✨ Add the `Zero.Companion.orNull(Byte)` function for Kotlin with tests, documentation and samples, then update the unreleased changelog.
- [ ] 🗑️ Deprecate the `Zero.Companion.fromByte(Byte)` function with an error level for using the `Zero.Companion.orThrow(Byte)` function instead, then update the unreleased changelog.
- [ ] 🗑️ Deprecate the `Zero.Companion.fromByteOrNull(Byte)` function with an error level for using the `Zero.Companion.orNull(Byte)` function instead, then update the unreleased changelog.
- [ ] 📝 Create an issue for removing the deprecated `Zero.Companion.fromByte(Byte)` and `Zero.Companion.fromByteOrNull(Byte)` functions in the next release.

**Comment by LVMVRQUXL** [comment]
Superseded by #686.
